librz/core/rop.c

```diff
@@ -1550,13 +1617,15 @@ RZ_API RzCmdStatus rz_core_rop_search(RZ_NONNULL RzCore *core, RZ_NONNULL RZ_OWN
  * Displays ROP gadgets from the gadgetSdb.
  * If unavailable, performs a ROP search with the input.
  */
-RZ_API RzCmdStatus rz_core_rop_gadget_info(RZ_NONNULL RzCore *core, RZ_NONNULL RZ_OWN RzRopSearchContext *context) {
+RZ_API RzCmdStatus rz_core_rop_gadget_info(RZ_NONNULL RzCore *core, RZ_NONNULL RZ_BORROW RzRopSearchContext *context) {
 	rz_return_val_if_fail(core && core->analysis && context, RZ_CMD_STATUS_ERROR);
 
 	if (!core->analysis->ht_rop_semantics) {
 		// TODO: resolve this logic later.
 	}
-	return rz_core_rop_search(core, context);
+	RzCmdStatus status = rz_core_rop_search(core, context);
+	rz_core_rop_search_context_free(context);
+	return status;
```

Comment:
here seems that context is owned not borrowed.

Reply:
Thanks for catching it!, changing this

Reply:
You forgot this one